server/cluster/cluster.go

```diff
@@ -380,30 +380,43 @@ func (c *RaftCluster) startGCTuner() {
 func (c *RaftCluster) runSyncConfig() {
 	defer logutil.LogPanic()
 	defer c.wg.Done()
-
 	ticker := time.NewTicker(time.Minute)
 	defer ticker.Stop()
+
 	stores := c.GetStores()
+	syncFunc := func() {
+		success, switchRaftV2Config := syncConfig(c.storeConfigManager, stores)
+		if switchRaftV2Config {
+			c.GetOpts().UseRaftV2()
+			if err := c.opt.Persist(c.GetStorage()); err != nil {
+				log.Warn("store config persisted failed", zap.Error(err))
+			}
+		}
+		if !success {
+			stores = c.GetStores()
+		}
+	}
 
-	syncConfig(c.storeConfigManager, stores)
+	syncFunc()
 	for {
 		select {
 		case <-c.ctx.Done():
 			log.Info("sync store config job is stopped")
 			return
 		case <-ticker.C:
-			if !syncConfig(c.storeConfigManager, stores) {
-				stores = c.GetStores()
-			}
+			syncFunc()
 		}
 	}
 }
 
-func syncConfig(manager *config.StoreConfigManager, stores []*core.StoreInfo) bool {
+// syncConfig syncs the config of the stores.
+// synced is true if sync config from one tikv.
+// switchRaftV2 is true if the config of tikv engine is changed and engine is raft-kv2.
```

Comment:
do we supported downgrade from kv2?

Reply:
no, raftkv2 is incompatible with raftkv1.